libvast/CMakeLists.txt

```diff
@@ -198,51 +198,30 @@ else ()
-    set(CAF_NO_AUTO_LIBCPP ON)
-    set(CAF_NO_OPENCL ON)
-    set(CAF_NO_EXAMPLES ON)
-    set(CAF_NO_UNIT_TESTS ON)
-    set(CAF_NO_PYTHON ON)
-    set(CAF_NO_TOOLS ON)
-    set(CAF_NO_SUMMARY ON)
-    if (NOT VAST_ENABLE_OPENSSL)
-      set(CAF_NO_OPENSSL TRUE)
-    endif ()
-    if (BUILD_SHARED_LIBS)
-      set(_linkage_suffix shared)
-    else ()
-      set(_linkage_suffix static)
-      set(CAF_BUILD_STATIC ON)
-      set(CAF_BUILD_STATIC_ONLY ON)
-    endif ()
+    set(CAF_ENABLE_EXAMPLES OFF)
+    set(CAF_ENABLE_TESTING OFF)
+    set(CAF_ENABLE_TOOLS OFF)
+    set(CAF_ENABLE_OPENSSL ${VAST_ENABLE_OPENSSL})
+    # add_subdirectory aux/caf checks if compiler supports the c++ 17. This check fails and the workaround
+    # can be removed once CAF cmake changes to use set(CMAKE_CXX_STANDARD) instead of it's own check
+    set(ORIGINAL_CXX_STANDARD ${CMAKE_CXX_STANDARD})
+    set(CMAKE_CXX_STANDARD 17)
     add_subdirectory(aux/caf)
-    VASTSystemizeTarget(libcaf_core_${_linkage_suffix})
-    set_target_properties(libcaf_core_${_linkage_suffix} PROPERTIES EXPORT_NAME
-                                                                    core)
-    add_library(CAF::core ALIAS libcaf_core_${_linkage_suffix})
-    target_compile_features(libcaf_core_${_linkage_suffix} INTERFACE cxx_std_17)
+    set(CMAKE_CXX_STANDARD ${ORIGINAL_CXX_STANDARD})
+    VASTSystemizeTarget(libcaf_core)
+    set_target_properties(libcaf_core PROPERTIES EXPORT_NAME core)
+    target_compile_features(libcaf_core INTERFACE cxx_std_17)
     target_compile_options(
-      libcaf_core_${_linkage_suffix}
+      libcaf_core
       PRIVATE -Wno-maybe-uninitialized -Wno-unqualified-std-cast-call
               -Wno-unknown-warning-option)
-    VASTSystemizeTarget(libcaf_io_${_linkage_suffix})
-    set_target_properties(libcaf_io_${_linkage_suffix} PROPERTIES EXPORT_NAME
-                                                                  io)
+    VASTSystemizeTarget(libcaf_io)
+    set_target_properties(libcaf_io PROPERTIES EXPORT_NAME io)
     target_compile_options(
-      libcaf_io_${_linkage_suffix}
-      PRIVATE -Wno-maybe-uninitialized -Wno-unqualified-std-cast-call
-              -Wno-unknown-warning-option)
-    find_package(Threads REQUIRED)
-    target_link_libraries(libcaf_io_${_linkage_suffix} Threads::Threads)
-    add_library(CAF::io ALIAS libcaf_io_${_linkage_suffix})
-    install(
-      TARGETS libcaf_core_${_linkage_suffix} libcaf_io_${_linkage_suffix}
-      EXPORT CAFTargets
-      ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT Development
-      LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT Runtime
-      RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" COMPONENT Runtime)
+      libcaf_io PRIVATE -Wno-maybe-uninitialized -Wno-unqualified-std-cast-call
+                        -Wno-unknown-warning-option)
     if (VAST_ENABLE_OPENSSL)
-      if (NOT TARGET libcaf_openssl_${_linkage_suffix})
+      if (NOT TARGET libcaf_openssl)
```

Comment:
Note that you will have a merge conflict here once #2719 gets merged, as that removes the `VAST_ENABLE_OPENSSL` option and makes that mandatory.